presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java

```diff
@@ -219,17 +219,20 @@ public boolean usesLegacyTableLayouts()
             unenforcedConstraint = clusteringPredicatesExtractor.getUnenforcedConstraints();
         }
 
-        if (handle.getPartitions().containsAll(partitionResult.getPartitions()) && handle.getClusteringKeyPredicates().equals(clusteringKeyPredicates)) {
+        Optional<List<CassandraPartition>> currentPartitions = handle.getPartitions();
+        if (currentPartitions.isPresent() &&
+                currentPartitions.get().containsAll(partitionResult.getPartitions()) &&
+                handle.getClusteringKeyPredicates().equals(clusteringKeyPredicates)) {
             return Optional.empty();
         }
 
         return Optional.of(
                 new ConstraintApplicationResult<>(new CassandraTableHandle(
                         handle.getSchemaName(),
                         handle.getTableName(),
-                        partitionResult.getPartitions(),
+                        Optional.of(partitionResult.getPartitions()),
                         clusteringKeyPredicates),
-                unenforcedConstraint));
+                        unenforcedConstraint));
```

Comment:
`Load list of partitions when no constraint was applied` commit

- it appears second on Github, but should be before "Skip" commit, as this is a prerequisite change.